### PR TITLE
Test/book model

### PR DESCRIPTION
**Pull request by floresv**
### Summary
* Add rspec, database-cleaner, faker, factor_bot, shoulda-matchers
* Setup rspec, configuration (database-cleaner, faker, factory_bot, shoulda-matchers) and add Book model tests

### Trello Card
https://trello.com/c/cxOiJRJs/23-crear-tests-para-el-modelo-de-book

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -42,6 +42,12 @@ group :development, :test do
   # Adds support for Capybara system testing and selenium driver
   gem 'capybara', '~> 2.13'
   gem 'selenium-webdriver'
+  # Tests
+  gem 'rspec-rails', '~> 3.4', '>= 3.4.2'
+  gem 'database_cleaner', '~> 1.5', '>= 1.5.3'
+  gem 'faker', '~> 1.6', '>= 1.6.6'
+  gem 'factory_bot_rails', '~> 4.8', '>= 4.8.2'
+  gem 'shoulda-matchers', '~> 3.1', '>= 3.1.1'
```
Comment (leamotta) [inline Gemfile]: for this gems you can just put them in a `group :test do` block, you won't really use them in a development environment

File: spec/factories/books.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+FactoryBot.define do
+  factory :book do
+    genre Faker::Book.genre
+    author Faker::Book.author
+    image Faker::Avatar.image
+    title Faker::Book.title
+    publisher Faker::Book.publisher
+    year Faker::Number.between(1950, Time.zone.today.year)
```
Comment (leamotta) [inline spec/factories/books.rb]: As a rule for better readability, we always put the Faker values between braces, and, for bonus points, all aligned.
`genre .{ Faker::Book.genre }` (without the `.`, markdown is not my thing :D)
`author { Faker::Book.author }`

File: spec/factories/books.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  end
+end
```
Comment (leamotta) [inline spec/factories/books.rb]: Also, the name of the factory file should be singular, like the model. You can choose between `book.rb` or `book_factory.rb`, the latter is better if you want to identify files quicker while searching in multiple files in VScode for example :)

File: spec/models/book_spec.rb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+describe Book do
+  subject(:book) { create(:book) }
+
+  it '#validate' do
+    is_expected.to validate_presence_of :genre
+    is_expected.to validate_presence_of :author
+    is_expected.to validate_presence_of :image
+    is_expected.to validate_presence_of :title
+    is_expected.to validate_presence_of :publisher
+    is_expected.to validate_presence_of :year
```
Comment (leamotta) [inline spec/models/book_spec.rb]: We usually have each of this validations inside a particular it, so it is easier to identify which one fails later. Also, we use parenthesis when using params.
`it { is_expected.to validate_presence_of(:genre) }`
`it { is_expected.to validate_presence_of(:author) }` and so on

File: spec/models/book_spec.rb [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  end
+
+  it 'valid book' do
+    is_expected.to be_valid
```
Comment (leamotta) [inline spec/models/book_spec.rb]: for this basic tests it is not necessary to add a description, you can just put
`it { is_expected.to be_valid }` and that's enough :)

**Review by leamotta — CHANGES_REQUESTED** [review]
Great! Just a few things to make things clearer and easier in the future

**Review by leamotta — COMMENTED** [review]
Besides, you need to tell Travis to run the tests. Add a bundle exec rspec spec -fd to the scripts

File: Gemfile (hunk 1 above)
Comment (floresv) [reply Gemfile]: That's True!

File: spec/factories/books.rb (hunk 2 above)
Comment (floresv) [reply spec/factories/books.rb]: Ok

File: spec/factories/books.rb (hunk 3 above)
Comment (floresv) [reply spec/factories/books.rb]: Ok

File: spec/models/book_spec.rb (hunk 4 above)
Comment (floresv) [reply spec/models/book_spec.rb]: Ok

File: spec/models/book_spec.rb (hunk 5 above)
Comment (floresv) [reply spec/models/book_spec.rb]: ohh Good.. I didn't know

**Review by leamotta — APPROVED** [review]
Awesome

**Pull request merged**